DancingGoat/Helpers/Extensions/HtmlHelperExtensions.cs

```diff
@@ -46,17 +48,19 @@ public static MvcHtmlString AssetImage(this HtmlHelper htmlHelper, Asset asset,
         /// <summary>
         /// Displays a <see cref="DateTime"/> in a formatted manner.
         /// </summary>
-        /// <param name="htmlHelper">HTML helper.</param>
+        /// <param name="htmlHelper">HTML helper</param>
+        /// <param name="expression">The expression of the model property</param>
         /// <param name="dateTime">The <see cref="DateTime"/> to format</param>
         /// <param name="format">The formatting character</param>
-        public static MvcHtmlString DateTimeFormatted(this HtmlHelper htmlHelper, DateTime? dateTime, string format)
+        /// <remarks>The TValue generic parameter is chosen instead of DateTime just to save views from falling to exceptions. With TValue, the views will get rendered, only this helper method will return an empty <see cref="MvcHtmlString"/>.</remarks>
+        public static MvcHtmlString DateTimeFormattedFor<TModel, TValue>(this HtmlHelper<TModel> htmlHelper, Expression<Func<TModel, TValue>> expression, string format)
         {
-            if (!dateTime.HasValue || string.IsNullOrEmpty(format))
+            if (expression.ReturnType != typeof(DateTime?) && expression.ReturnType != typeof(DateTime))
             {
                 return MvcHtmlString.Empty;
             }
 
-            return MvcHtmlString.Create(dateTime.Value.ToString(format));
+            return htmlHelper.DisplayFor(expression, "DateTime", new { FormatCharacter = format });
```

Comment:
I've created a simple `DateTimeFormatterParameters` class to pass to the DisplayFor (instead of an anonymous type). Then, I use `nameof` in DateTime.cshtml. Hopefully a good solution ...